x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/snapshot_policies/snapshot_policies.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiButtonIcon,
+  EuiCallOut,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { useLoadSnapshotPolicies } from '../../../../services/api';
+
+interface Props {
+  value: string;
+  onChange: (value: string) => void;
+}
+export const SnapshotPolicies: React.FunctionComponent<Props> = ({ value, onChange }) => {
+  const { error, isLoading, data, sendRequest } = useLoadSnapshotPolicies();
+
+  const policies = data.map((name: string) => ({
+    label: name,
+    value: name,
+  }));
+
+  const onComboChange = (options: EuiComboBoxOptionOption[]) => {
+    if (options.length > 0) {
+      onChange(options[0].label);
+    } else {
+      onChange('');
+    }
+  };
+
+  const onCreateOption = (newValue: string) => {
+    onChange(newValue);
+  };
+
+  let calloutContent;
+  if (error) {
+    calloutContent = (
+      <Fragment>
+        <EuiSpacer size="m" />
+        <EuiCallOut
+          data-test-subj="policiesErrorCallout"
+          size="s"
+          iconType="help"
+          color="warning"
+          title={
+            <Fragment>
+              <FormattedMessage
+                id="xpack.indexLifecycleMgmt.editPolicy.deletePhase.noPoliciesLoadedTitle"
+                defaultMessage="Couldn't load existing policies."
+              />
+
+              <EuiButtonIcon
+                size="s"
+                color="warning"
+                onClick={sendRequest}
+                iconType="refresh"
+                aria-label={i18n.translate(
+                  'xpack.indexLifecycleMgmt.editPolicy.deletePhase.reloadPoliciesLabel',
+                  {
+                    defaultMessage: 'Reload policies',
+                  }
+                )}
+              />
+            </Fragment>
+          }
+        >
+          <FormattedMessage
+            id="xpack.indexLifecycleMgmt.editPolicy.deletePhase.noPoliciesLoadedMessage"
+            defaultMessage="You still can type in a policy name."
```

Comment:
```suggestion
            defaultMessage="You still can enter a policy name to search for."
```

Reply:
For this case, I am currently disabling options list in the combo box, so that it functions as a simple text field. The user can type in any value, but I'd like to encourage them to make sure they input a name of an existing snapshot policy. Maybe add a link to "Snapshot and Restore" here too?

Reply:
Hmm...if the list of policies failed to load, we should instruct users to reload the list and retry their search for an existing policy. Does clicking the reload button make a call to get all existing policies? Are there any other actions they can take beyond reload + retry? 

>**Unable to load existing policies**
>Refresh this field and enter the name of an existing snapshot policy.

Reply:
yes, currently there is a "refresh" icon button to "reload policies". It sends a request to the ES API in hopes of getting snapshot policies list. All snapshot policies are loaded at the same time, if the request succeeds. And I believe there is not much the user can do, if this request fails. 
<img width="446" alt="Screenshot 2020-07-08 at 18 00 37" src="https://user-images.githubusercontent.com/6585477/86942026-f32d7180-c144-11ea-9bad-b575096ea35b.png">